src/handler.js

```diff
@@ -251,7 +251,7 @@ module.exports = class Handler {
     if (pushInfo.revision) {
       job.origin.revision = pushInfo.revision;
     } else {
-      job.origin.revision_hash = pushInfo.revision_hash;
+      job.origin.revision = pushInfo.revision_hash;
```

Comment:
@djmitche : It's hard for me to tell what exactly is coming from upstream here, tbh.  We no longer use ``revision_hash`` at all.  So this change to not set that field of ``job.origin`` is correct.  But I don't think you guys should be getting a ``revision_hash`` either.  

Just for context, the ``revision_hash`` used to be a hash built up of all the revisions for the push.  But now we always use the "tip" revision.  As I'm sure you do.  I would check where ``pushInfo`` might get a ``revision_hash``.  Hopefully, if it DOES get one, it will actually be the tip revision and all will be well.  But maybe it'd be better to just ensure that ``pushInfo`` never gets a ``revision_hash`` field.

Reply:
I suppose my point here is that, if ``revision_hash`` is somehow, indeed a hash of all the commits for a push, then the ``job`` in question here will be orphaned, because it will never find a matching ``push``.

Reply:
If the job has a ``revision`` that is not the tip of a ``push`` we already have, then it will be stuck in our rabbitMQ queue till we've retried 10 times, and then be dropped.  It'll never get added to Treeherder.  So, I agree, just dropping the old task would be best so as not to waste time on it.  :)